server/src/main/java/org/elasticsearch/common/util/concurrent/PrioritizedEsThreadPoolExecutor.java

```diff
@@ -95,14 +95,13 @@ public TimeValue getMaxTaskWaitTime() {
 
     private void addPending(Collection<Runnable> runnables, List<Pending> pending, boolean executing) {
         for (Runnable runnable : runnables) {
-            if (runnable instanceof TieBreakingPrioritizedRunnable) {
-                TieBreakingPrioritizedRunnable t = (TieBreakingPrioritizedRunnable) runnable;
-                Runnable innerRunnable = t.runnable;
+            if (runnable instanceof TieBreakingPrioritizedRunnable tieBreaking) {
+                PrioritizedRunnable innerRunnable = tieBreaking.runnable;
                 if (innerRunnable != null) {
                     /** innerRunnable can be null if task is finished but not removed from executor yet,
                      * see {@link TieBreakingPrioritizedRunnable#run} and {@link TieBreakingPrioritizedRunnable#runAndClean}
                      */
-                    pending.add(new Pending(super.unwrap(innerRunnable), t.priority(), t.insertionOrder, executing));
+                    pending.add(new Pending(innerRunnable, tieBreaking.priority(), tieBreaking.insertionOrder, executing));
```

Comment:
I'm not sure if it's safe to remove the `unwrap` call here, and also don't know how to check if your change is valid.

Reply:
Yes it is safe - in the case of the `PrioritizedEsThreadPoolExecutor` none of the `PrioritizedRunnable` implementations are also `WrappedRunnable`.

Even if they were - it wouldn't matter because of the way the pending instances are used.

Reply:
To add - I wasn't sure if https://github.com/crate/crate/blob/253533b236d570cefaf7be6ead0d6555c9370502/server/src/main/java/org/elasticsearch/common/util/concurrent/PrioritizedEsThreadPoolExecutor.java#L142 is safe - but if it weren't `getPending` would already have triggered an assertion error in those cases.

Would be easy enough to support that case, but should first see why/where it is required.